src/main/webapp/main/Routes.js

```diff
@@ -28,6 +30,8 @@ const Routes = ( props ) => (
     <Route path= {"/dataUseLetter/show"} render = {(routeProps) =>  <DataUseLetterIndex {...routeProps} {...props}/> }/>
     <Route path= {"/submissions/add-new"} render={(routeProps) => <SubmissionForm {...routeProps} {...props} />} />
     <Route path= {"/dataUse/show"} render = {(routeProps) =>  <DataUseRestrictionDetails {...routeProps} {...props}/> }/>
+    <Route path= {"/dataUse/restriction"} render = {(routeProps) =>  <DataUseRestrictionEdit {...routeProps} {...props}/> }/>
+    <Route path= {"/*"} render = {(routeProps) =>  <PageNotFound {...routeProps} {...props}/> }/>
```

Comment:
The not found component is a recent addition to this PR. Has `path= {"/*"}` been tested? I'm think it won't have any effect on the non-react pages, but we should be sure.

Reply:
This is intended to work in combination with urlMappings change. Anything not resolved in urlMappings is redirected to be handled by Routes. And if not found there a 404 page is shown. It was tested late yesterday and worked as expected, with a still not resolved issue of getting an additional 500 error in notFound page. This change is related to handle yor comment about "redirectToMainContainer" .

Reply:
The conversation around `redirectToMainContainer` was related to having three different methods performing the same function in a Groovy controller method. I don't see the connection to this. In either case, it's a relatively small concern.

Reply:
Will try to explain it. We had 3 methods performing same function on Groovy controller because were handling 3 different urls (vies, pages). Since those pages are implemented in React now, but called from Grails by now, we need to go to the Grails controller and redirect to React Router which will handle the request at the end. While the 3 methods looked the same, redirectToMainContainer call was different on each case because the payload: the requested URL. That requested URL ended in Routes and was handled by Switch and Router. So, we removed actions from Grails controller and add a "generic" urlmapping in urlMappings to handle any not-matcher url and redirect it to React Router. Since any non existing url will end in react router as well, we added the /* route at end to catch all non exiting pages. That is the idea. Not sure if the best, at least the best we had. :-)

Reply:
I think this might be the disconnect ...

> Since those pages are implemented in React now, but called from Grails by now

If we have pages implemented in React, then there should be no calls from Grails to those pages. It sounds like there's a circular route handling process here that could use some de-convoluting. I wish I could spend some time simplifying this ... just not possible right now.

Reply:
I agree, there is something that needs improvement. We already have some stories in this Sprint for this, converting remaining pages/views to React.